Sources/StreamChatUI/Utils/AudioPlayer/AudioPlayerObserving.swift

```diff
@@ -0,0 +1,132 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import AVFoundation
+import Foundation
+
+/// Describes an object that can provide AVPlayer related updates.
+public protocol AudioPlayerObserving {
```

Comment:
No action for now, just food for thought: If we decide reusing this component in SwiftUI, we should probably also expose publishers.